internal/03_publish_demo.md

```diff
@@ -3,86 +3,136 @@ publish
 Lucy D’Agostino McGowan
 5/3/2017
 
--   [check published status](#check-published-status)
--   [publish it](#publish-it)
+-   [Motivation](#motivation)
+-   [Push a file into a Sheet](#push-a-file-into-a-sheet)
+-   [Check publication status (should be FALSE)](#check-publication-status-should-be-false)
+-   [get URL](#get-url)
+-   [switch to different account](#switch-to-different-account)
+-   [this shouldn't work](#this-shouldnt-work)
+-   [publish it on Drive](#publish-it-on-drive)
+-   [try again!](#try-again)
 -   [clean up](#clean-up)
 
-This is a little demo to show how we can check if a file is published & publish it if we so desire.
+Motivation
+----------
 
-``` r
-library('dplyr')
-library('googledrive')
-```
+Push a table into a Sheet.
+
+Try to read it *as another user* Assume you even have the key. You will fail.
+
+Now, as the user who owns the Sheet, publish it.
+
+Now, as the other user, try again to read it via googlesheets. You should succeed.
+
+Push a file into a Sheet
+------------------------
 
 ``` r
-write.table("This is a little demo", "demo.txt")
-drive_upload("demo.txt", name = "Happy Little Demo")
+drive_auth("drive-token.rds")
+drive_chickwts <- drive_upload("~/desktop/chickwts.csv", type = "spreadsheet")
 ```
 
     ## File uploaded to Google Drive: 
-    ## demo.txt 
-    ## As the Google document named:
-    ## Happy Little Demo
+    ## ~/desktop/chickwts.csv 
+    ## As the Google spreadsheet named:
+    ## chickwts
+
+Check publication status (should be FALSE)
+------------------------------------------
 
 ``` r
-my_file <- drive_get_id("Happy Little Demo") %>%
-  drive_file
+drive_check_publish(drive_chickwts)
 ```
 
-check published status
-----------------------
+    ## The latest revision of the Google Drive file 'chickwts' is not published.
+
+get URL
+-------
 
 ``` r
-my_file %>%
-  drive_check_publish
+url <- drive_share_link(drive_chickwts)
 ```
 
-    ## The latest revision of the Google Drive file 'Happy Little Demo' is not published.
+``` r
+key <- drive_chickwts$id
+```
 
-the `drive_check_publish` also outputs a new file with a tibble called `publish` with publication information. We can overwrite `my_file` to see this
+switch to different account
+---------------------------
 
 ``` r
-my_file <- my_file %>%
-  drive_check_publish
+gs_auth("sheets-token.rds")
 ```
 
-    ## The latest revision of the Google Drive file 'Happy Little Demo' is not published.
+this shouldn't work
+-------------------
 
 ``` r
-my_file$publish
+try(gs_url(url, lookup = FALSE))
```

Comment:
I think you should possibly be setting visibility to private here. I mean, we will still want to demonstrate an error, but for the right reasons.